packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts

```diff
@@ -164,8 +175,25 @@ function trackLongestInteractions(getViewInteractionCount: () => number) {
 export function trackViewInteractionCount(viewLoadingType: ViewLoadingType) {
   initInteractionCountPolyfill()
   const previousInteractionCount = viewLoadingType === ViewLoadingType.INITIAL_LOAD ? 0 : getInteractionCount()
+  let viewInteractionCount = 0
+  let stopped = false
+
+  function computeViewInteractionCount() {
+    viewInteractionCount = getInteractionCount()! - previousInteractionCount
+  }
+
   return {
-    getViewInteractionCount: () => getInteractionCount()! - previousInteractionCount,
+    getViewInteractionCount: () => {
+      if (!stopped) {
+        computeViewInteractionCount()
+      }
+
+      return viewInteractionCount
+    },
+    stopViewInteractionCount: () => {
+      computeViewInteractionCount()
+      stopped = true
+    },
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** having a typed "state" would make the code a bit more explicit (the link between `viewInteractionCount` and `stopped` would be clearer and `computeViewInteractionCount` would not have sideeffect):

```ts
  type State = { stopped: false } | { stopped: true, interactionCount: number }
  let state: State = { stopped: false }

  function computeViewInteractionCount() {
    return getInteractionCount()! - previousInteractionCount
  }

  return {
    getViewInteractionCount: () => {
      if (state.stopped) {
        return state.interactionCount
      }
      return computeViewInteractionCount()
    },
    stopViewInteractionCount: () => {
      state = { stopped: true, interactionCount: computeViewInteractionCount() }
    },
```